packages/apollo-cli/src/Config.ts

```diff
@@ -0,0 +1,184 @@
+import fs from 'node:fs'
+
+import Joi from 'joi'
+import YAML from 'yaml'
+
+import { ConfigError } from './utils.ts'
+
+interface RootCredentials {
+  username?: string
+  password?: string
+}
+
+interface Profile {
+  address?: string
+  accessType?: string
+  accessToken?: string
+  rootCredentials?: RootCredentials
+}
```

Comment:
Could you change the type for the Profile to be this:

```ts
interface BaseProfile {
  address: string
  accessType: 'google' | 'microsoft' | 'guest'
  token: string
}

interface RootProfile extends Omit<BaseProfile, 'accessType'> {
  accessType: 'root'
  rootCredentials: {
    username: string
    password: string
  }
}

export type Profile = BaseProfile | RootProfile
```

This way we can depend on "address" and "accessType" not being undefined, and "rootCredential" being defined only if "accessType" is "root".